internal/pkg/cri/pods.go

```diff
@@ -182,18 +178,22 @@ func stopAndRemove(ctx context.Context, stopAction StopAction, client *Client, p
 		container := container // https://golang.org/doc/faq#closures_and_goroutines
 
 		g.Go(func() error {
-			log.Printf("%s container %s/%s:%s", action, pod.Metadata.Namespace, pod.Metadata.Name, container.Metadata.Name)
-
 			// TODO(andrewrynhard): Can we set the timeout dynamically?
-			if criErr := client.StopContainer(ctx, container.Id, 30); criErr != nil {
-				if grpcstatus.Code(criErr) == codes.NotFound {
-					return nil
-				}
+			if container.State == runtimeapi.ContainerState_CONTAINER_RUNNING || container.State == runtimeapi.ContainerState_CONTAINER_UNKNOWN {
```

Comment:
do we have any potential other states in between as reported by cri, maybe a container is still restarting/not healthy?

Reply:
these are two different states, there's a sandbox (pod) state and a container state. 

we are in fact interested in stopping containers, as they hold the mounts, etc. 

we stop the pods as well, but that mostly means tearing down CNI stuff